internal/ld/ld.go

```diff
@@ -389,11 +392,59 @@ func (b BranchRep) TotalHunkCount() int {
 	return count
 }
 
+func (b BranchRep) WriteToCSV(outDir, projKey, repo, sha string) (path string, err error) {
+	tag := b.Name
+	if len(sha) >= 7 {
+		tag = sha[:7]
+	}
+
+	absPath, err := validation.NormalizeAndValidatePath(outDir)
+	if err != nil {
+		return "", fmt.Errorf("invalid outDir '%s': %s", outDir, err)
+	}
+	path = filepath.Join(absPath, fmt.Sprintf("coderefs_%s_%s_%s.csv", projKey, repo, tag))
+
+	f, err := os.Create(path)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	w := csv.NewWriter(f)
+	records := make([][]string, 0, len(b.References)+1)
+	for _, ref := range b.References {
+		records = append(records, ref.toRecords()...)
+	}
+
+	// sort csv by flag key
+	sort.Slice(records, func(i, j int) bool {
+		// sort by flagKey -> path -> startingLineNumber
+		for k := 0; k < 3; k++ {
+			if records[i][k] != records[j][k] {
+				return records[i][k] < records[j][k]
+			}
+		}
```

Comment:
This loop is not exactly my favorite way to sort, but I think it's safe to assume the ordering of these values are invariant